src/main/java/org/folio/search/service/ResourceService.java

```diff
@@ -142,6 +149,36 @@ private <T> List<T> getEventsThatCanBeIndexed(List<T> events, Function<T, String
     return eventsToIndex;
   }
 
+  /**
+   * There may be a case when some data is moved between instances.
+   * In such case old and new fields of the event will have different instanceId.
+   * This method will create 2 events out of 1 and erase 'old' field in an original event.
+   * */
+  private List<ResourceEvent> extractEventsForDataMove(List<ResourceEvent> resourceEvents) {
+    if (resourceEvents == null) {
+      return Collections.emptyList();
+    }
+
+    return resourceEvents.stream()
+      .flatMap(resourceEvent -> {
```

Comment:
@viacheslavkol, what do you think about filtering by non-instance resource type, to avoid the above processing for instance events?

Reply:
I'm not sure what exactly do you meant.
If you are talking about ResourceEventType (ResourceEvent.type) - then data passed into this method has been already filtered (grouped by INDEX operation).
If you meant ResourceEvent.resourceName - then this method is used only for events that are all converted to instance events in Kafka listener so I think this is not necessary.